elide-async/src/main/java/com/yahoo/elide/async/export/formatter/CSVExportFormatter.java

```diff
@@ -64,15 +75,46 @@ public String format(PersistentResource resource, Integer recordNumber) {
         return str.toString();
     }
 
-    private String resourceToJSON(PersistentResource resource) {
+    /**
+     * Generate CSV Header when Observable is Empty.
+     * @param projection EntityProjection object.
+     * @return returns Header string which is in CSV format.
+     */
+    protected Observable<String> generateCSVHeader(EntityProjection projection) {
+        Observable<String> header = Observable.empty();
+        Iterator itr = projection.getAttributes().iterator();
         StringBuilder str = new StringBuilder();
-
-        try {
-            str.append(mapper.writeValueAsString(resource.getObject()));
-        } catch (JsonProcessingException e) {
-            log.error("Exception when converting to JSON {}", e.getMessage());
-            throw new IllegalStateException(e);
+        int columnCount = 0;
+        while (itr.hasNext()) {
```

Comment:
I think this can be written in more readable fashion as a stream:

```java
projection.getAttributes().stream().
   .map(attr -> //convert to name of the attribute)
   .map(//append quotes)
   .collect(Collectors.joining(",")
```